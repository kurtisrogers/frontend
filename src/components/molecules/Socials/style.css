nav.socials a {
  display: flex;
  color: var(--brand-white);
}

nav.socials a:focus {
  color: var(--brand-black);
}

nav.socials ul {
  padding: 0px;
  list-style-type: none;
}

nav.socials a:hover {
  filter: drop-shadow(2px 2px 2px currentColor);
}
