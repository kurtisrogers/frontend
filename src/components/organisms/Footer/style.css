footer {
  background: var(--brand-yellow);
  background: var(--brand-primary-gradient);
  padding: 4rem 1rem;
}

footer .filterPanel--blur {
  display: flex;
  justify-content: space-between;
  max-width: var(--brand-base-grid);
  margin: 0 auto;
}

footer .logo {
  display: block;
  margin-bottom: 0.625rem;
}

footer nav:not(.socials) {
  text-align: right;
}

footer nav ul {
  display: block;
  margin: 0px;
}

footer nav:not(.socials) li {
  list-style-type: none;
  margin-bottom: 0.5rem;
}

footer nav a {
  display: inline-flex;
  color: var(--brand-black);
  padding: 0px;
  outline: none;
  font-weight: 600;
}

footer div.filterPanel > :first-child {
  width: calc(100% - 4rem);
}

.footer .footer__socials {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

footer .socials ul {
  display: flex;
  flex-direction: row;
  margin: 0px;
}