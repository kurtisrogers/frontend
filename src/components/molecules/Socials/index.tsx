import Icon from "@/components/atoms/Icon";
import "./style.css";

// icon data (dumb component)
import { linkedin } from "@/data/icons";

export default function Socials() {
  return (
    <nav class="socials">
      <ul>
        <li>
          <a href="https://www.linkedin.com/in/roguetrufflepig/" target="_blank">
            <Icon width={50} height={50} viewbox="0 0 24 24" icondata={linkedin} />
            <span class="sr-only">A link to my LinkedIn profile (opens a new tab)</span>
          </a>
        </li>
      </ul>
    </nav>
  );
}
