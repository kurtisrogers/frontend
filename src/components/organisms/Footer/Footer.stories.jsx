import Footer from "./index";
import "./style.css";

export default {
  title: "Organisms/Footer",
  component: Footer,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/solid/writing-docs/docs-page
  tags: ["autodocs"],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/solid/configure/story-layout
    layout: "fullscreen"
  }
};

const Template = () => <Footer />;

export const Default = Template.bind({});
// Default.args = {
//   // Add any props you want to pass to the Header component
// };
