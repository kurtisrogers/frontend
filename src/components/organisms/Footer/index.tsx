import { JSX } from "solid-js";
import Logo from "@/components/atoms/Logo";
import Socials from "@/components/molecules/Socials";
import CSSFilterPanel from "@/components/atoms/CSSFilterPanel";
import "./style.css";

interface Props {
  children?: JSX.Element;
}

export default function Footer({ children }: Readonly<Props>) {
  return (
    <footer tabindex="-1">
      <div class="footer">
        <CSSFilterPanel variant="blur">
          <div class="footer__socials">
            <Logo variant="secondary" />
            <Socials />
          </div>
          {/* some other stuff here? */}
          {children}
        </CSSFilterPanel>
      </div>
    </footer>
  );
}
