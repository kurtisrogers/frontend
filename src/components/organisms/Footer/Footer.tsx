import "./Footer.css";

export default function Footer(props: any) {
  return (
    <footer>
      <p>Hello I'm the footer</p>
      {props.children}
    </footer>
  )
}
