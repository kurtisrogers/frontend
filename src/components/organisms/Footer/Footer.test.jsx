import { render } from "@solidjs/testing-library";
import Footer from ".";
import Socials from "../../molecules/Socials";
import { describe, it, expect } from "vitest";

vi.mock(import("@solidjs/router"), async (importOriginal) => {
  const actual = await importOriginal()
  return {
    ...actual,
    A: (props) => <a {...props} data-testid="mock-link">{props.children}</a>
  }
})

describe("<Footer />", () => {
  it("renders correctly and has a link to my linkedin profile", () => {
    const { getByText } = render(() => (
      <Footer>
        <Socials />
      </Footer>
    ));
    const linkButton = getByText("A link to my LinkedIn profile (opens a new tab)");

    expect(linkButton).toMatchSnapshot();
    expect(linkButton).toBeInTheDocument();
  });
});
