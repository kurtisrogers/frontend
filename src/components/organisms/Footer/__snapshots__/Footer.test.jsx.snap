// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer /> > renders correctly and has a link to my linkedin profile 1`] = `
<span
  class="sr-only"
>
  A link to my LinkedIn profile (opens a new tab)
</span>
`;
